doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1506,160 +1494,140 @@ Managing the local context
+             eassert ( @ident := @term )
+             eassert @one_term {? @as_ipat } {? by @ltac_expr3 }
+      :name: eassert; _; _
 
-.. tacv:: pose proof (@ident := @term)
+      Unlike :tacn:`assert`, the :n:`@type`\s, :n:`@term`\s and :n:`@one_term`\s in
+      :tacn:`eassert` may contain :term:`holes <hole>`, denoted by :n:`_`,
+      for which the tactic will create existential variables.  This lets you
+      avoid specifying the asserted statement completely before starting to
+      prove it.
 
-   This is an alternative syntax for :n:`assert (@ident := @term)` and
-   :n:`pose proof @term as @ident`, following the model of :n:`pose
-   (@ident := @term)` but dropping the value of :token:`ident`.
+.. tacn:: enough ( @ident : @type ) {? by @ltac_expr3 }
+          enough @one_term {? @as_ipat } {? by @ltac_expr3 }
+   :name: enough; _
 
-.. tacv:: epose proof (@ident := @term)
+   .. todo PR The only difference vs assert is where the goal is added?  If so, then
+      this entry should only refer to assert and mention the difference(s).
+      There are several similar tactics in this section.  Maybe we should add a
+      summary of their differences?  Perhaps you could briefly outline them?
 
-   This is an alternative syntax for :n:`eassert (@ident := @term)`
-   and :n:`epose proof @term as @ident`, following the model of
-   :n:`epose (@ident := @term)` but dropping the value of
-   :token:`ident`.
+   The first form adds a new hypothesis :n:`@ident : @type` to the context
+   and adds :n:`@type` as a new subgoal after the current goal (rather than
+   before it as :tacn:`assert` would).  If present, :n:`@ltac_expr3` must solve the
+   initial goal entirely after the extra assumption :n:`@type` is added and possibly
+   destructed; otherwise, the tactic will fail.
 
-.. tacv:: enough (@ident : @type)
-   :name: enough
+   In the second form, if :n:`@as_ipat` isn't specified, the tactic adds a new
+   hypothesis with a name chosen by Coq.  Otherwise, it transforms :n:`@type`
+   as specified by :n:`@as_ipat` and adds the resulting new hypotheses.
+   If the :n:`as @simple_intropattern` clause generates more than one subgoal,
+   :n:`@ltac_expr3`, if present, is applied to all of them.
 
-   This adds a new hypothesis of name :token:`ident` asserting :token:`type` to the
-   goal the tactic :tacn:`enough` is applied to. A new subgoal stating :token:`type` is
-   inserted after the initial goal rather than before it as :tacn:`assert` would do.
+   .. tacn:: eenough ( @ident : @type ) {? by @ltac_expr3 }
+             eenough @one_term {? @as_ipat } {? by @ltac_expr3 }
+      :name: eenough; _
 
-.. tacv:: enough @type
+      Unlike :tacn:`enough`, the :n:`@type`\s and :n:`@one_term`\s in
+      :tacn:`eenough` may contain :term:`holes <hole>`, denoted by :n:`_`,
+      for which the tactic will create existential variables.  This lets you
+      avoid specifying the asserted statement completely before starting to
+      prove it.
```

Comment:
```suggestion
      avoid specifying the asserted statement completely.
```

The issue with the last part of the sentence is that it may suggest that the next step is to prove the asserted statement, whereas in the case of `enough`, the next step is normally to prove that this additional hypothesis is enough (thus the tactic name) to prove the initial goal.